ofrak_core/test_ofrak/unit/test_ofrak_server.py

```diff
@@ -454,29 +454,110 @@ async def test_find_and_replace(ofrak_client: TestClient, hello_world_elf):
     assert resp.status == 200
 
 
+# Returns (comment range count, comment count)
+# comment range count = # of unique Optional[Range]s comments are mapped to
+# comment count = total # of comments across all Optional[Range]s
+def get_comment_count(resp_body_json) -> Tuple[int, int]:
+    attributes = resp_body_json["modified"][0]["attributes"]
+    comment_range_count = 0
+    comment_count = 0
+
+    for attr_array in attributes:
+        if attr_array[0] == "ofrak.core.comments.CommentsAttributes":
+            comment_data = attr_array[1][1]["comments"]
+            for kv_pair in comment_data:
+                comment_range_count += 1
+                comment_count += len(kv_pair[1])
+
+    return (comment_range_count, comment_count)
+
+
 async def test_add_comment(ofrak_server, aiohttp_client, hello_world_elf):
     client = await aiohttp_client(ofrak_server._app)
     create_resp = await client.post(
         "/create_root_resource", params={"name": "test"}, data=hello_world_elf
     )
     create_body = await create_resp.json()
-    resp = await client.post(f"/{create_body['id']}/add_comment", json=[[0, 425], "test"])
+    # Try creating comments on invalid ranges
+    resp = await client.post(
+        f"/{create_body['id']}/add_comment",
+        json=[[0, len(hello_world_elf) + 1], "test comment out of bounds"],
+    )
+    assert resp.status == 500
+    resp = await client.post(
+        f"/{create_body['id']}/add_comment",
+        json=[[-1, len(hello_world_elf)], "test comment out of bounds"],
+    )
+    assert resp.status == 500
```

Comment:
It's probably a bit brittle to assert that this returns a 500 (internal server) error. It would likely be better to assert that this does not return a 200.